src/theme-default/components/SwitchAppearance/index.tsx

```diff
@@ -1,17 +1,24 @@
 import { Switch } from '../Switch';
-import React from 'react';
+import { useEffect } from 'react';
 import styles from './index.module.scss';
 import SunSvg from './sun.svg';
 import MoonSvg from './moon.svg';
 import { ComponentPropsWithIsland } from '../../../shared/types';
+import { inBrowser } from '../../../runtime/utils';
 
 export function SwitchAppearance(_props: ComponentPropsWithIsland) {
+  useEffect(() => {
+    applyDefaultAppearance();
+  }, []);
```

Comment:
There is a problem using `useEffect` because it has been finished first render when executing `applyDefaultAppearance`.So the first screen will suddenly flash.

Reply:
You can try in https://island-git-fork-godliangcy-feat-theme-default-sanyuan0704.vercel.app/

Reply:
Thanks for correcting. And I think it's nice to let it to be closed in #27